core/src/test/java/com/novoda/merlin/service/MerlinServiceShould.java

```diff
@@ -112,6 +117,32 @@ public void callMerlinNetworkListenerOnDisconnectedWhenAConnectedConnectivityEve
         verify(rxCallbacksManager).onDisconnect();
     }
 
+    @Test
+    public void updateCurrentNetworkStatusRetrieverWhenSetHostNameIsCalled() {
+        String newTestHostName = "someNewTestHostName";
+        Context context = mock(Context.class);
+        ContextWrapper contextWrapper = mock(ContextWrapper.class);
+        ConnectivityManager connectivityManager = mock(ConnectivityManager.class);
+        Mockito.when(contextWrapper.getApplicationContext()).thenReturn(context);
+        Mockito.when(context.getSystemService(Context.CONNECTIVITY_SERVICE)).thenReturn(connectivityManager);
+
+        MerlinService merlinService = new TestDoubleMerlinServiceReturningMockingContext(contextWrapper);
+
+        merlinService.onCreate();
+        merlinService.setHostname(newTestHostName, new ResponseCodeValidator() {
+            @Override
+            public boolean isResponseCodeValid(int responseCode) {
+                return true;
+            }
+        });
+
+        CurrentNetworkStatusRetriever currentNetworkStatusRetriever = (CurrentNetworkStatusRetriever) Whitebox.getInternalState(merlinService, "currentNetworkStatusRetriever");
+        HostPinger hostPinger = (HostPinger) Whitebox.getInternalState(currentNetworkStatusRetriever, "hostPinger");
+        String currentTestHostAddress = (String) Whitebox.getInternalState(hostPinger, "hostAddress");
```

Comment:
I think this might be a bit of a smell (testing the internals of the class), shouldn't we try to test the _what_ rather than the _how_? (testing what would be the result of resetting the status retriever rather than testing that we reset the retriever)